app/components/draft/draft_post/draft_message/index.tsx

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useState} from 'react';
+import {ScrollView, View, useWindowDimensions, type LayoutChangeEvent} from 'react-native';
+import Animated from 'react-native-reanimated';
+
+import ShowMoreButton from '@app/components/post_list/post/body/message/show_more_button';
+import {useTheme} from '@app/context/theme';
+import {useShowMoreAnimatedStyle} from '@app/hooks/show_more';
+import {getMarkdownBlockStyles, getMarkdownTextStyles} from '@app/utils/markdown';
+import {makeStyleSheetFromTheme} from '@app/utils/theme';
+import {typography} from '@app/utils/typography';
+import Markdown from '@components/markdown';
+
+import type {DraftModel} from '@app/database/models/server';
+import type {UserMentionKey} from '@typings/global/markdown';
+
+type Props = {
+    draft: DraftModel;
+    layoutWidth: number;
+    location: string;
+}
+
+const EMPTY_MENTION_KEYS: UserMentionKey[] = [];
+const SHOW_MORE_HEIGHT = 54;
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        messageContainer: {
+            width: '100%',
+        },
+        reply: {
+            paddingRight: 10,
+        },
+        message: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 200),
+        },
+        pendingPost: {
+            opacity: 0.5,
+        },
+    };
+});
+
+const DraftMessage: React.FC<Props> = ({
+    draft,
+    layoutWidth,
+    location,
+}) => {
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+    const blockStyles = getMarkdownBlockStyles(theme);
+    const textStyles = getMarkdownTextStyles(theme);
+    const [height, setHeight] = useState<number|undefined>();
+    const [open, setOpen] = useState(false);
+    const dimensions = useWindowDimensions();
+    const maxHeight = Math.round((dimensions.height * 0.5) + SHOW_MORE_HEIGHT);
+    const animatedStyle = useShowMoreAnimatedStyle(height, maxHeight, open);
+
+    const onLayout = useCallback((event: LayoutChangeEvent) => setHeight(event.nativeEvent.layout.height), []);
+    const onPress = () => setOpen(!open);
+
+    return (
+        <>
+            <Animated.View style={animatedStyle}>
+                <ScrollView
+                    keyboardShouldPersistTaps={'always'}
+                    scrollEnabled={false}
+                    showsVerticalScrollIndicator={false}
+                    showsHorizontalScrollIndicator={false}
+                >
```

Comment:
When the content is wrapped in a ScrollView, React Native can measure the full height of the content accurately, even if the content is overflowing. This is because ScrollView enables scrolling, so it doesn't limit the height calculation to what's immediately visible. Without ScrollView, React Native would only measure the visible part, often resulting in a shorter, incorrect height measurement. This may result in not showing the ShowMore button.